BootloaderCommonPkg/Library/FileSystemLib/FileSystemLib.c

```diff
@@ -83,17 +83,32 @@ InitFileSystem (
   )
 {
   EFI_STATUS   Status;
+  UINT32       Type;
+
+  Status = EFI_UNSUPPORTED;
+  if (mCurrentFsType == EnumFileSystemMax) {
+    Type = EnumFileSystemTypeFat;
+    if (FixedPcdGet32 (PcdSupportedFileSystemMask) & (1 << Type)) {
+      mFileSystemFuncs[Type].InitFileSystem     = FatInitFileSystem;
+      mFileSystemFuncs[Type].GetFileByName      = FatGetFileByName;
+    }
 
-  Status = EFI_INVALID_PARAMETER;
-  if ((FsType == EnumFileSystemTypeFat) || (FsType == EnumFileSystemTypeAuto)) {
-    Status = FatInitFileSystem (SwPart, PartHandle, FsHandle);
+    Type = EnumFileSystemTypeExt2;
+    if (FixedPcdGet32 (PcdSupportedFileSystemMask) & (1 << Type)) {
+      mFileSystemFuncs[Type].InitFileSystem     = ExtInitFileSystem;
+      mFileSystemFuncs[Type].GetFileByName      = ExtGetFileByName;
+    }
   }
 
-  if (EFI_ERROR (Status)) {
-    if ((FsType == EnumFileSystemTypeExt2) || (FsType == EnumFileSystemTypeAuto)) {
-      Status = ExtInitFileSystem (SwPart, PartHandle, FsHandle);
+  for (Type = EnumFileSystemTypeFat; Type < EnumFileSystemMax; Type++) {
+    if (((FsType & Type) != 0) && (mFileSystemFuncs[Type].InitFileSystem != NULL)) {
+      mCurrentFsType = Type;
```

Comment:
better to move this to after line 100 so that the init will only do one time.

Reply:
will update the code